tutorials/introductory/usage.py

```diff
@@ -5,59 +5,56 @@
 
 This tutorial covers some basic usage patterns and best-practices to
 help you get started with Matplotlib.
+"""
 
-.. _general_concepts:
-
-General Concepts
-================
-
-:mod:`matplotlib` has an extensive codebase that can be daunting to many
-new users. However, most of matplotlib can be understood with a fairly
-simple conceptual framework and knowledge of a few important points.
-
-Plotting requires action on a range of levels, from the most general
-(e.g., 'contour this 2-D array') to the most specific (e.g., 'color
-this screen pixel red'). The purpose of a plotting package is to assist
-you in visualizing your data as easily as possible, with all the necessary
-control -- that is, by using relatively high-level commands most of
-the time, and still have the ability to use the low-level commands when
-needed.
-
-Therefore, everything in matplotlib is organized in a hierarchy. At the top
-of the hierarchy is the matplotlib "state-machine environment" which is
-provided by the :mod:`matplotlib.pyplot` module. At this level, simple
-functions are used to add plot elements (lines, images, text, etc.) to
-the current axes in the current figure.
-
-.. note::
+# sphinx_gallery_thumbnail_number = 4
+import matplotlib.pyplot as plt
+import numpy as np
 
-   Pyplot's state-machine environment behaves similarly to MATLAB and
-   should be most familiar to users with MATLAB experience.
+##############################################################################
+#
+# A simple example
+# ================
+#
+# Matplotlib graphs your data on `~.figure.Figure`\s (i.e., windows, Jupyter
+# widgets, etc.), each of which can contain one or more `~.axes.Axes` (i.e., an
+# area where points can be specified in terms of x-y coordinates (or theta-r
+# in a polar plot, or x-y-z in a 3D plot, etc.).  In the following example, we
+# create a figure using the `.pyplot.figure` function, an axes on that figure
+# using the `.Figure.add_subplot` method, and graph some data on that axes
+# using the `.Axes.plot` method:
 
-The next level down in the hierarchy is the first level of the object-oriented
-interface, in which pyplot is used only for a few functions such as figure
-creation, and the user explicitly creates and keeps track of the figure
-and axes objects. At this level, the user uses pyplot to create figures,
-and through those figures, one or more axes objects can be created. These
-axes objects are then used for most plotting actions.
+fig = plt.figure()  # Create a figure.
```

Comment:
I tend to use `fig, ax = plt.subplots()` here.

Pro:
- it's less cluttered and simpler to remember than two functions.
- it's the canonical way to create a single Axes and Figure nowadays.

Con:
- There's less emphasis on the figure + axes concepts.

Reply:
I would rather leave the two-way approach:
- it showcases the figure/axes distinction better.
- `plt.subplots()` is currently canonical because its is much older than `Figure.subplots()` (I think it's ~2.0 vs ~1.3), but I would actually like to slowly push towards making `plt.figure().subplots()` the canonical way and have `plt.subplots()` be considered a pyplot wrapper (just like `plt.plot()`).  Really, I think there's two functions in pyplot that are "special": `figure()` (create a figure and register it to pyplot) and `show()` (spin up the event loop); everything else is helpers.

Not going to overly insist on that, though.

Reply:
There's good reason for both. Let's see if anybody else has an opinion.

Reply:
Not a fan of `plt.figure().subplots()`; it doesn't actually return the figure handle, so 90% of the time you will need to do:

```python
fig = plt.figure()
axs = fig.subplots()
```

Reply:
It's much rarer to need a handle on the figure, and even then, if you only have a single axes, you can just get it via `ax.figure`.

In any case, I'm fine with switching to `plt.subplots()` for this specific tutorial, but then you get to rewrite the corresponding comments to explain what's happening.

Reply:
I've rewritten to `plt.subplots()` and pushed as a separate commit.